pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -135,12 +119,14 @@ func (t ResolvedPipelineRunTask) IsStarted() bool {
 
 func (t *ResolvedPipelineRunTask) checkParentsDone(facts *PipelineRunFacts) bool {
 	stateMap := facts.State.ToMap()
-	// check if parent tasks are done executing,
-	// if any of the parents is not yet scheduled or still running,
-	// wait for it to complete before evaluating when expressions
 	node := facts.TasksGraph.Nodes[t.PipelineTask.Name]
 	for _, p := range node.Prev {
-		if !stateMap[p.Task.HashKey()].IsDone() {
+		if stateMap[p.Task.HashKey()].TaskRun == nil {
+			if t.Skip(facts) {
+				continue
+			}
+		}
+		if !stateMap[p.Task.HashKey()].IsSuccessful() && !stateMap[p.Task.HashKey()].IsFailure() {
 			return false
 		}
 	}
```

Comment:
Note that this is changed to be consistent with `checkTaskDone`: https://github.com/tektoncd/pipeline/blob/1314c25422ae07bc5993e1c7710507432f0425c6/pkg/reconciler/pipelinerun/resources/pipelinerunstate.go#L304-L324